packages/tools/client-debugger/client-debugger-view/src/components/TelemetryView.tsx

```diff
@@ -104,10 +117,121 @@ export function TelemetryView(): React.ReactElement {
 		};
 	}, [messageRelay, setTelemetryEvents]);
 
+	/**
+	 * Interface for each item in the telemetry table.
+	 */
+	interface Item {
+		category: string;
+		eventName: string;
+		information: string;
+	}
+
+	const items: Item[] =
+		telemetryEvents !== undefined
+			? telemetryEvents?.map((message) => {
+					return {
+						category: message.logContent.category,
+						eventName: message.logContent.eventName,
+						information: JSON.stringify(
+							message.logContent,
+							jsonSerializationTransformer,
+							"  ",
+						),
+					};
+			  })
+			: [];
+
+	const columns: TableColumnDefinition<Item>[] = [
+		createTableColumn<Item>({
+			columnId: "category",
+			renderHeaderCell: () => {
+				return <b>Category</b>;
+			},
+			renderCell: (message) => {
+				return <TableCellLayout>{message.category}</TableCellLayout>;
+			},
+		}),
+		createTableColumn<Item>({
+			columnId: "eventName",
+			renderHeaderCell: () => {
+				return <b>Event Name</b>;
+			},
+			renderCell: (message) => {
+				return (
+					<div style={{ maxWidth: 160, whiteSpace: "normal", wordWrap: "break-word" }}>
+						{message.eventName.slice("fluid:telemetry:".length)}
+					</div>
+				);
+			},
+		}),
+		createTableColumn<Item>({
+			columnId: "information",
+			renderHeaderCell: () => {
+				return <b>Information</b>;
+			},
+			renderCell: (message) => {
+				// eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
I think you need to hoist this up to the component root (per the linter rule), or you'll want to extract the cell rendering logic into its own component. As is, I believe this violates React state requirements.

Reply:
Actually, I might be wrong there. I guess it depends on how the table component is calling `renderCell`. If it's being rendered like a React component, then this is effectively just an inline component, in which case there isn't a problem. Even still, probably best to extract the logic into its own component to be safe (and so we don't have to disable our linter rules).